robot_framework/subprocess/masseoprettelse_nova.py

```
@@ -0,0 +1,147 @@
+        name = _get_name_from_cpr(cpr = queue_element.reference, nova_access=nova_access, cases=cases)
+        queue_element_count[0] += 1
+        try:
+            case = (_find_matching_case(data_dict["Sagsoverskrift"], cases)
+                    if data_dict["Brug eksisterende sag"] == "Valgt"
+                    else _create_case(queue_element.reference, name, data_dict, nova_access))
+
+            caseworker = config.CASEWORKER
+            nova_notes.add_text_note(
+                case.uuid,
+                data_dict["Notat overskrift"],
+                data_dict["Notat tekst"],
+                caseworker,
+                True,
+                nova_access)
+        except LookupError:
+            orchestrator_connection.set_queue_element_status(queue_element.id, QueueStatus.FAILED, f"Sagsoverskrift '{data_dict['Sagsoverskrift']}' ikke fundet.")
+        except HTTPError as e:
+            orchestrator_connection.set_queue_element_status(queue_element.id, QueueStatus.FAILED, json.loads(e.response.text)["title"])
+        # Set status Done for this note and look for the next queue element
+        else:
+            orchestrator_connection.set_queue_element_status(queue_element.id, QueueStatus.DONE)
+
+
+def _get_name_from_cpr(cpr: str, nova_access: NovaAccess, cases: list[NovaCase]) -> str:
+    """Find name from lookup by address, and if not found (such as when using test CPRs) do a lookup in cases.
+
+    Args:
+        cpr: ÍD of the person we are looking for.
+        nova_access: A token to access the KMD Nova API.
+
+    Returns:
+        The name of the person with the provided CPR.
+    """
+    address = nova_cpr.get_address_by_cpr(cpr, nova_access)
+    if address:
+        return address['name']
+    for case in cases:
+        for case_party in case.case_parties:
+            if case_party.identification == cpr and case_party.name:
+                return case_party.name
+    return "No Name"
```

This should probably raise an error